engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/TypeDefinitionSet.scala

```diff
@@ -0,0 +1,47 @@
+package pl.touk.nussknacker.engine
+
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import org.springframework.expression.spel.{ExpressionState}
+import org.springframework.expression.spel.ast.TypeReference
+import pl.touk.nussknacker.engine.api.Context
+import pl.touk.nussknacker.engine.api.expression.ExpressionParseError
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedClass, TypingResult}
+import pl.touk.nussknacker.engine.definition.{DefinitionExtractor, ProcessDefinitionExtractor, TypeInfos}
+import pl.touk.nussknacker.engine.spel.TypedNode
+import pl.touk.nussknacker.engine.spel.ast.SpelAst.RichSpelNode
+import pl.touk.nussknacker.engine.spel.internal.EvaluationContextPreparer
+
+
+object TypeDefinitionSet {
+
+  def empty: TypeDefinitionSet = apply(Set.empty)
+
+  def apply(typeDefinitionSet: Set[TypeInfos.ClazzDefinition]): TypeDefinitionSet = {
+
+//    val clazzDefinitionSet = typeDefinitionSet.map(
+//      clazzDefinition => clazzDefinition.clazzName)
+
+    new TypeDefinitionSet(typeDefinitionSet)
+  }
+}
+
+class TypeDefinitionSet(typeDefinitions: Set[TypeInfos.ClazzDefinition]) {
+
+//  def typeDefinitionMap = typeDefinitions
+
+  def validateTypeReference(typeReference: TypeReference, evaluationContextPreparer: EvaluationContextPreparer): Validated[NonEmptyList[ExpressionParseError], TypedClass] = {
+
+    val evaluationContext = evaluationContextPreparer.prepareEvaluationContext(Context(""), Map.empty)
+
+    val typeReferenceClazz = typeReference.getValue(new ExpressionState(evaluationContext))
```

Comment:
1. And what about reference to not existing class?
2. Also I wonder if invoking `getValue` on AST during typing is a good choice. I see that it mutate this instance - taka a look at spel's implementation. Can you verify if such muatating is safe (won't cause that something will broke up in serialization or during real evaluation)?

Reply:
1. Think I fixed, please take a look
2. I analysed the problem and this mutating seems to be safe, added a comment describing it